src/fetchers/default-fetcher.nix

```diff
@@ -11,6 +11,7 @@
   defaultPackageVersion,
   sourceOverrides,
   sources,
+  sourceRoot ? null,
```

Comment:
I don't think we should put much effort into maintaining downward compatibility. The dream2nix project is in early stage development and marked as `WIP`. Anyone depending on it should count with breaking changes on any update.
It unnecessarily increases the complexity of our code and we have no clear policy on when to remove the extra code.

But if you prefer to have it like this, I'm fine with it as well.

Reply:
It being optional should allow people to call default fetcher without a `sourceRoot`. This should help if someone only uses the `dream-lock.json` file to build it, without passing a "source root" (because they know there is none). For example `dream2nix-index-lib` relies on being able to only build from the dream-lock files, so if this was required it wouldn't work.